auxiliary/producers/replayer/replayer.go

```diff
@@ -0,0 +1,267 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+	pb "github.com/bptlab/cepta/models/grpc/replayer"
+	libdb "github.com/bptlab/cepta/osiris/lib/db"
+	kafkaproducer "github.com/bptlab/cepta/osiris/lib/kafka/producer"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	tspb "github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/jinzhu/gorm"
+	"github.com/sirupsen/logrus"
+)
+
+// EventData types
+type EventData interface {
+	GetActualTime() time.Time
+	ToEvent() proto.Message
+	GetTopic() string
+}
+
+// Any types can be everything
+type Any interface {
+}
+
+// Replayer implements the Producer interface
+type Replayer struct {
+	Ctrl       chan pb.InternalControlMessageType
+	TableName  string
+	MustMatch  *[]string
+	Timerange  *pb.Timerange
+	SortColumn string
+	Limit      *int
+	Speed      *int32
+	Active     *bool
+	Repeat     bool
+	Mode       *pb.ReplayType
+	Offset     int
+	Db         *libdb.DB
+	Brokers    []string
+	Log        *logrus.Logger
+	running    bool
+	producer   *kafkaproducer.KafkaProducer
+}
+
+// Producer can be used to extract data from a database
+type Producer interface {
+	GetParent() *Replayer
+	Start() error
+}
+
+func toTimestamp(t time.Time) *tspb.Timestamp {
+	ts, _ := ptypes.TimestampProto(t)
+	return ts
+}
+
+// TODO: rename for better understanding
+func buildQuery(column string, timerange *pb.Timerange) []string {
+	frmt := "2006-01-02 15:04:05" // Go want's this date!
+	query := []string{}
+	if start, err := ptypes.Timestamp(timerange.GetStart()); err != nil && start.Unix() > 0 {
+		query = append(query, fmt.Sprintf("%s >= '%s'", column, start.Format(frmt)))
+	}
+	if end, err := ptypes.Timestamp(timerange.GetEnd()); err != nil && end.Unix() > 0 {
+		query = append(query, fmt.Sprintf("%s < '%s'", column, end.Format(frmt)))
+	}
+	return query
+}
+
+func (this Replayer) enrichQuery(query *gorm.DB) *gorm.DB {
```

Comment:
"receiver name should be a reflection of its identity; don't use generic names such as "this" or "self""

This is the warning from the official golint plugin. You should never use "this" or "self" in go!